frontend/ui/widget/textboxwidget.lua

```diff
@@ -112,7 +121,14 @@ function TextBoxWidget:_splitCharWidthList()
         else
             -- Backtrack the string until the length fit into one line.
             local c = self.char_width_list[idx].char
-            if util.isSplitable(c) then
+            -- Only split on spaces (so punctuation sticks to prev word) and
+            -- only if space is not followed by some specific punctuation,
+            -- e.g. "word :" or "word )" where we don't want to split on
+            -- space if next_c = ":"  (in french, there is a space before
+            -- a colon, and it better not be wrapped there).
+            local next_c = idx+1 <= size and self.char_width_list[idx+1].char or "willnotbefound"
+            -- was : if util.isSplitable(c) then
+            if c == " " and not non_splittable_space_tailers:find(next_c, 1, true) then
```

Comment:
this breaks line breaking for CJK. If you look at the definition of `util.isSplitable`. I suggest you pass the next_c as a optional second argument to `util.isSplitable` and move the logic there. Also it's better to set `next_c` as `nil` if `c` is at the end of the line.